### PR TITLE
feat(postgresql): Add Support For Different PK Types

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

depends on: https://github.com/cloudquery/plugin-sdk/pull/732

Initial implementation, still needs testing!

Related to: https://github.com/cloudquery/cloudquery/issues/8811

### CODE REVIEW

File: plugins/destination/postgresql/client/migrate.go [hunk 1]
```diff
@@ -377,6 +374,29 @@ func (c *Client) createTableIfNotExist(ctx context.Context, table *schema.Table)
...
 }
 
+func (c *Client) listPKs(table *schema.Table) []string {
+	// If PKS are not needed for the writeMode then just return early
+	if !c.enabledPks() {
+		return []string{}
+	}
+	pks := []string{}
+	if c.spec.PKMode == specs.PKModeCQID {
+		pks = append(pks, schema.CqIDColumn.Name)
```
Comment (erezrokah) [inline plugins/destination/postgresql/client/migrate.go:384]: I'm not sure destinations should reference CqID directly. Update the table PKs detentions in the SDK based on the spec PK mode, WDYT?

Comment (bbernays) [reply plugins/destination/postgresql/client/migrate.go:384]: If we don't reference the `CqID` how do we enable users to be able to switch between `composite-keys` and `cq-ids`?

Comment (erezrokah) [reply plugins/destination/postgresql/client/migrate.go:384]: See my comment here https://github.com/cloudquery/plugin-sdk/pull/732#pullrequestreview-1339843474
We can update the column options here https://github.com/cloudquery/plugin-sdk/blob/ac55bec5f3c06b3cdd9a0409394d5bf5cf059137/plugins/source/plugin.go#L282 for example

Comment (erezrokah) [reply plugins/destination/postgresql/client/migrate.go:384]: If we do it on destinations, we can override the PK definitions here https://github.com/cloudquery/plugin-sdk/blob/ac55bec5f3c06b3cdd9a0409394d5bf5cf059137/plugins/destination/plugin.go#L187

Comment (bbernays) [reply plugins/destination/postgresql/client/migrate.go:384]: Great call! Thanks for the heads up

**Comment by bbernays** [comment]
Closing this, as it can be done 100% handled in the SDK (thank you @erezrokah for that tip!)

**Pull request closed without merging**